wellregistry/registry/admin/bulk_upload.py

```diff
@@ -38,13 +38,23 @@ def _get_state_lookup(country, state_name):
 def _get_county_lookup(country, state, county_name):
     if not country or not state:
         return None
-    qs = CountyLookup.objects.filter(county_nm=county_name, state_id=state, country_cd=country)
+    qs = CountyLookup.objects.filter(
+        county_nm=county_name, state_id=state, country_cd=country)
     if len(qs) == 0:
         return None
     return qs[0]
 
 
-def _get_monitoring_location(row, user):
+def _validate_decimal(field_name, dec_value, row_index, warning_messages):
```

Comment:
I understand why you are doing this here but it results in two messages (confusingly) when validating the bulk upload. You might want to read up on validation in the Django docs. I think what we want is field level validation first before creating the monitoring_location object.

Reply:
As a philosophical discussion, inputting the data via a CSV or a form or anything else requires validation. Django validation can still be used in such cases (and we do a full_clean to validate multi field validations). I just saw this quote on a Django list serv: "Do not fight the framework by reinventing the wheel, use it." The context was different then ours (user permissions in that case) but I think applicable. Having said that what I really object to here is creating this new message class "warning_messages" which results in two messages generated for a single error on a required decimal field. However, if Candice is fine with the result then I think we can go forward.